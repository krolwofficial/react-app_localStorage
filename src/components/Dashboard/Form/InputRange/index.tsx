import React from "react";
//import { debounce } from "lodash";
//import _ from "lodash";

const RangeInput: any = ({ text, max, handleChange }) => (
  <div className="input-field col s12">
    <p>Select {text}: </p>
    <p className="range-field">
      <input
        className="form__input"
        type="range"
        min="0"
        max={max}
        // onChange={({ target: { value } }) =>
        //   _.debounce(handleChange(value), 550)
        // }
        onChange={({ target: { value } }) => handleChange(value)}
      />
    </p>
  </div>
);

export default RangeInput;
