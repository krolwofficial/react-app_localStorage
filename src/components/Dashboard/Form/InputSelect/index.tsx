import React from "react";
import M from "materialize-css";
import { useDispatch } from "react-redux";

const SelectInput: any = ({ handleChange }) => {
  const dispatch = useDispatch();

  React.useEffect(() => {
    // init the MaterializeCSS <select> styling
    let elem = document.querySelector("select");
    M.FormSelect.init(elem);
  }, []);

  return (
    <div className="input-field col s12">
      <p>Select color: </p>
      <select
        onChange={({ target: { value } }) => handleChange(value)}
        defaultValue={"red"}
      >
        <option value="red">Red</option>
        <option value="orange">Orange</option>
        <option value="yellow">Yellow</option>
        <option value="green">Green</option>
        <option value="purple">Purple</option>
        <option value="teal">Teal</option>
        <option value="aqua">Aqua</option>
        <option value="blue">Blue</option>
        <option value="black">Black</option>
        <option value="gray">Gray</option>
      </select>
    </div>
  );
};

export default SelectInput;
