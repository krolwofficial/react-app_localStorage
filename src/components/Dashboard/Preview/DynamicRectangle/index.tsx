import React from "react";
import { useSelector } from "react-redux";

const DynamicRectangle: React.FC = () => {
  const rectangle = useSelector<any, any>(state => state.rectangle);

  const styles = {
    div: {
      width: `${rectangle.width}px`,
      height: `${rectangle.height}px`,
      borderRadius: `${rectangle.borderRadius}%`,
      background: rectangle.background
    }
  };

  return <div style={styles.div} />;
};

export default DynamicRectangle;
