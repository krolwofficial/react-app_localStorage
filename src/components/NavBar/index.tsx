import React, { useState } from "react";
import MenuButton from "./MenuButton";
import cn from "classnames";
import { NavLink } from "react-router-dom";

const NavBar: React.FC = () => {
  const [open, setOpen] = useState<boolean>(false);

  return (
    <nav
      className={cn({
        "nav-wrapper": true,
        open: open
      })}
    >
      <ul className="nav-wrapper__ul">
        <li className="nav-wrapper__li">
          <NavLink
            exact={true}
            activeClassName="is-active"
            to="/"
            className="nav-wrapper__a"
            onClick={() => setOpen(false)}
          >
            Dashboard
          </NavLink>
        </li>
        <li className="nav-wrapper__li">
          <NavLink
            activeClassName="is-active"
            to="/gallery"
            className="nav-wrapper__a"
            onClick={() => setOpen(false)}
          >
            Gallery
          </NavLink>
        </li>
      </ul>
      <MenuButton handleClick={() => setOpen(!open)} />
    </nav>
  );
};

export default NavBar;
