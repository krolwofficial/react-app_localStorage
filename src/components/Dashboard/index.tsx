import React from "react";
import Form from "./Form";
import Preview from "./Preview";

const Dashboard = () => (
  <div className="row page ">
    <div className="col s12 m5 l5 offset-l1">
      <div className="container">
        <Form />
      </div>
    </div>
    <div className="col s12 m7 l6">
      <div className="container white">
        <Preview />
      </div>
    </div>
  </div>
);

export default Dashboard;
