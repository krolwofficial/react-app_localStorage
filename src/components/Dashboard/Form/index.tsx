import React from "react";
import InputRange from "./InputRange";
import InputSelect from "./InputSelect";
import { withRouter, RouteComponentProps } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { RectangleProps, TRectangle } from "../../../types/components";
import uuid from "uuid/v4";
import {
  AddRectangleAction,
  SetWidthRectangle,
  SetHeightRectangle,
  SetBackgroundRectangle,
  SetBRadiusRectangle
} from "../../../types/store";

const Form: React.FC<RouteComponentProps> = ({ history }) => {
  const dispatch = useDispatch();
  const rectangle = useSelector<RectangleProps, TRectangle>(
    state => state.rectangle
  );

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>): any => {
    e.preventDefault();
    dispatch<AddRectangleAction>({
      type: "ADD_RECTANGLE",
      payload: {
        id: uuid(),
        width: rectangle.width,
        height: rectangle.height,
        background: rectangle.background,
        borderRadius: rectangle.borderRadius
      }
    });
    history.push("/gallery");
  };

  return (
    <div className="col s12 white dashboard">
      <div className="container">
        <form onSubmit={e => handleSubmit(e)} className="form">
          <InputRange
            handleChange={val =>
              dispatch<SetWidthRectangle>({
                type: "SET_WIDTH",
                payload: val
              })
            }
            max={200}
            text={"width"}
          />
          <InputRange
            handleChange={val =>
              dispatch<SetHeightRectangle>({
                type: "SET_HEIGHT",
                payload: val
              })
            }
            max={200}
            text={"height"}
          />
          <InputSelect
            handleChange={val =>
              dispatch<SetBackgroundRectangle>({
                type: "SET_BACKGROUND",
                payload: val
              })
            }
          />
          <InputRange
            handleChange={val =>
              dispatch<SetBRadiusRectangle>({
                type: "SET_BRADIUS",
                payload: val
              })
            }
            max={50}
            text={"border radius"}
          />
          <input
            type="submit"
            className=" form__input btn btn-med red"
            value="Save"
          />
        </form>
      </div>
    </div>
  );
};

export default withRouter(Form);
