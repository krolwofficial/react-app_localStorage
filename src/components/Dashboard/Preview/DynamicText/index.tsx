import React from "react";
import { useSelector } from "react-redux";
import { RectangleProps, TRectangle } from "../../../../types/components";

const PreviewText: React.FC = () => {
  const rectangle = useSelector<RectangleProps, TRectangle>(
    state => state.rectangle
  );

  return (
    <ul>
      <li>rectangle's width = {rectangle.width} px</li>
      <li>rectangle's height = {rectangle.height} px</li>
      <li>rectangle's color = {rectangle.background}</li>
      <li>rectangle's border radius = {rectangle.borderRadius} %</li>
    </ul>
  );
};

export default PreviewText;
