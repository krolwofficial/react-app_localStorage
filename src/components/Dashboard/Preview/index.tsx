import React from "react";
import DynamicRectangle from "./DynamicRectangle";
import DynamicText from "./DynamicText";

const Preview: React.FC = () => (
  <div className="col s12 white dashboard preview">
    <div className="container">
      <p className="preview__title">Preview:</p>
      <DynamicText />
    </div>
    <div className="preview__box">
      <DynamicRectangle />
    </div>
  </div>
);

export default Preview;
