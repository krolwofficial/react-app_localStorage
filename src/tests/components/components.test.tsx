import React from "react";
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import InputRange from "../../components/Dashboard/Form/InputRange";
import InputSelect from "../../components/Dashboard/Form/InputSelect";
import Form from "../../components/Dashboard/Form";
import Preview from "../../components/Dashboard/Preview";
import Dashboard from "../../components/Dashboard";
import NavBar from "../../components/NavBar";

Enzyme.configure({ adapter: new EnzymeAdapter() });

test("Should render Dashboard component", () => {
  const wrapper = shallow(<Dashboard />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("Should render Form component", () => {
  const wrapper = shallow(<Form />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("Should render Preview component", () => {
  const wrapper = shallow(<Preview />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("Should render InputRange component", () => {
  const wrapper = shallow(
    <InputRange handleChange={() => []} max={200} text={"width"} />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("Should render InputSelect component", () => {
  const wrapper = shallow(<InputSelect handleChange={() => []} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("Should render NavBar component", () => {
  const wrapper = shallow(<NavBar />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
