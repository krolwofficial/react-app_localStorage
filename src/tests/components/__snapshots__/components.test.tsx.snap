// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Dashboard component 1`] = `
<div
  className="row page "
>
  <div
    className="col s12 m5 l5 offset-l1"
  >
    <div
      className="container"
    >
      <withRouter(Form) />
    </div>
  </div>
  <div
    className="col s12 m7 l6"
  >
    <div
      className="container white"
    >
      <Preview />
    </div>
  </div>
</div>
`;

exports[`Should render Form component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Should render InputRange component 1`] = `
<div
  className="input-field col s12"
>
  <p>
    Select 
    width
    : 
  </p>
  <p
    className="range-field"
  >
    <input
      className="form__input"
      defaultValue="40"
      max={200}
      min="0"
      onChange={[Function]}
      type="range"
    />
  </p>
</div>
`;

exports[`Should render InputSelect component 1`] = `
<div
  className="input-field col s12"
>
  <p>
    Select color: 
  </p>
  <select
    defaultValue="red"
    onChange={[Function]}
  >
    <option
      value="red"
    >
      Red
    </option>
    <option
      value="orange"
    >
      Orange
    </option>
    <option
      value="yellow"
    >
      Yellow
    </option>
    <option
      value="green"
    >
      Green
    </option>
    <option
      value="purple"
    >
      Purple
    </option>
    <option
      value="teal"
    >
      Teal
    </option>
    <option
      value="aqua"
    >
      Aqua
    </option>
    <option
      value="blue"
    >
      Blue
    </option>
    <option
      value="black"
    >
      Black
    </option>
    <option
      value="gray"
    >
      Gray
    </option>
  </select>
</div>
`;

exports[`Should render NavBar component 1`] = `
<nav
  className="nav-wrapper"
>
  <ul
    className="nav-wrapper__ul"
  >
    <li
      className="nav-wrapper__li"
    >
      <NavLink
        activeClassName="is-active"
        className="nav-wrapper__a"
        exact={true}
        onClick={[Function]}
        to="/"
      >
        Dashboard
      </NavLink>
    </li>
    <li
      className="nav-wrapper__li"
    >
      <NavLink
        activeClassName="is-active"
        className="nav-wrapper__a"
        onClick={[Function]}
        to="/gallery"
      >
        Gallery
      </NavLink>
    </li>
  </ul>
  <MenuButton
    handleClick={[Function]}
  />
</nav>
`;

exports[`Should render Preview component 1`] = `
<div
  className="col s12 white dashboard preview"
>
  <div
    className="container"
  >
    <p
      className="preview__title"
    >
      Preview:
    </p>
    <PreviewText />
  </div>
  <div
    className="preview__box"
  >
    <DynamicRectangle />
  </div>
</div>
`;
